### PR TITLE
feat: add login in popup options to config popup width and weigth

### PR DESCRIPTION
**Pull request by mario-mui**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
Hey :wave: thanks for this.

I think that we'd probably prefer not to go down this route of standardising some config for this just yet, I think that we would probably want you to either provide your own `Oauth2` implementation instead by copying most of this https://github.com/backstage/backstage/blob/714a317e0deae7c4f0d266137d4ef088fbe02f00/packages/core-app-api/src/apis/implementations/auth/oauth2/OAuth2.ts and passing it in as a plugin.

Alternatively, we could maybe extend https://github.com/backstage/backstage/blob/714a317e0deae7c4f0d266137d4ef088fbe02f00/packages/core-app-api/src/apis/implementations/auth/oauth2/OAuth2.ts#L40 these options to pass in some `popupOptions` which you can provide yourself when you create the `factory`.

Don't think that it's worth adding this to config and doing it for something that's quite edge casey is my reason for pushing back a little bit.

**Comment by mario-mui** [comment]
> 

Sure, I had change this solution with Extending `OAuth2CreateOptions`. When backstage adopter want to config their login-in popup size they need to create an api factory.

ps: I also think use config is more simple to app developer 😂

**Comment by benjdlambert** [comment]
@mario-mui yep - it sure is easier, but I think that this is a little to specific to be putting in config at the moment, I'd rather open it up for interpretation so that you can get this value from anywhere instead of being super opinionated about this config at the moment. We've also got some steps with #18372 that we probably want to settle first that might change how we think about this a little bit too.

**Comment by benjdlambert** [comment]
Hey :wave: sorry think this looks better now! Wanna get another signoff from another maintainer though first @backstage/maintainers. They're mostly out on holiday at the minute, so might have to wait a week or so, but will try and get some feedback asap!

**Comment by mario-mui** [comment]
> Hey 👋 sorry think this looks better now! Wanna get another signoff from another maintainer though first @backstage/maintainers. They're mostly out on holiday at the minute, so might have to wait a week or so, but will try and get some feedback asap!

Got it

File: packages/core-app-api/src/lib/AuthConnector/DefaultAuthConnector.ts
Comment (Rugvip) [inline packages/core-app-api/src/lib/AuthConnector/DefaultAuthConnector.ts:59]:
```suggestion
   * Options used to configure auth popup
```

File: packages/core-app-api/src/lib/AuthConnector/types.ts
Comment (Rugvip) [inline packages/core-app-api/src/lib/AuthConnector/types.ts:37]: Setting width or height to 0 will also cause it to go fullscreen, perhaps that's enough? Although thinking a bit more about it I feel that these should perhaps be groped a bit. What do you think about the following?

```ts
interface PopupOptions {
  size?: { width: number; height: number } | { fullscreen: true }
}
```

That way you avoid a situation where only width or height is set, or all 3 are set.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you, makes sense! 👍 

Nit + some thinking around the API

**Review by Rugvip — APPROVED** [review]
That'll also do it, thanks! 👍 

Let's :shipit:

**Pull request merged**
